src/common/coap_libcoap.cpp

```diff
@@ -181,7 +181,7 @@ otbrError AgentLibcoap::Send(Message &       aMessage,
         VerifyOrExit(pdu->length + sizeof(meta) < pdu->max_size, errno = EMSGSIZE);
 
         tid = coap_send_confirmed(&mCoap, mCoap.endpoint, &remote, pdu);
-        memcpy(pdu->hdr + pdu->length, &meta, sizeof(meta));
+        memcpy((unsigned char *)pdu->hdr + pdu->length, &meta, sizeof(meta));
```

Comment:
nit: use `static_cast<uint8_t*>(pdu->hdr)`?

Reply:
I used (unsigned char *) because this is how it's implemented in libcoap in exactly the same cases.